Sparkle/SUSignatures.m

```diff
@@ -0,0 +1,138 @@
+    return [[NSData alloc] initWithBase64Encoding:stripped];
+}
+
+- (instancetype)initWithDsa:(NSString * _Nullable)maybeDsa ed:(NSString * _Nullable)maybeEd25519
+{
+    self = [super init];
+    if (self) {
+        if (maybeDsa != nil) {
+            _dsaSignature = decode(maybeDsa);
+        }
+        if (maybeEd25519 != nil) {
+            NSData *data = decode(maybeEd25519);
+            assert(64 == sizeof(self->ed25519_signature));
+            [data getBytes:self->ed25519_signature length:sizeof(self->ed25519_signature)];
+        }
+    }
+    return self;
+}
+
+- (const unsigned char *)ed25519Signature {
+// Xcode may enable this in pedantic mode
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdirect-ivar-access"
+    for(size_t i=0; i < sizeof(self->ed25519_signature); i++) {
+        if (self->ed25519_signature[i] != 0) {
+            return self->ed25519_signature;
+        }
+    }
+    return NULL;
+#pragma clang diagnostic pop
+}
+
+- (instancetype)initWithCoder:(NSCoder *)decoder
+{
+    self = [super init];
+    if (self) {
+        NSData *dsaSignature = [decoder decodeObjectOfClass:[NSData class] forKey:SUDSASignatureKey];
+        if (dsaSignature) {
+            _dsaSignature = dsaSignature;
+        }
+
+        NSData *edSignature = [decoder decodeObjectOfClass:[NSData class] forKey:SUEDSignatureKey];
+        if (edSignature) {
+            if (edSignature.length != sizeof(self->ed25519_signature)) {
+                return nil;
+            }
+            [edSignature getBytes:self->ed25519_signature];
+        }
+    }
+    return self;
+}
+
+- (void)encodeWithCoder:(NSCoder *)coder
+{
+    if (self.dsaSignature) {
+        [coder encodeObject:self.dsaSignature forKey:SUDSASignatureKey];
+    }
+    if (self.ed25519Signature) {
+// Xcode may enable this in pedantic mode
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdirect-ivar-access"
+        NSData *edSignature = [NSData dataWithBytesNoCopy:&self->ed25519_signature length:sizeof(self->ed25519_signature) freeWhenDone:false];
+#pragma clang diagnostic pop
+        [coder encodeObject:edSignature forKey:SUEDSignatureKey];
+    }
+}
+
++ (BOOL)supportsSecureCoding {
+    return YES;
+}
+
+@end
+
+@implementation SUPublicKeys
+@synthesize dsaPubKey = _dsaPubKey;
+
+- (instancetype)initWithDsa:(NSString * _Nullable)maybeDsa ed:(NSString * _Nullable)maybeEd25519
+{
+    self = [super init];
+    if (self) {
+        _dsaPubKey = maybeDsa;
+        if (maybeEd25519 != nil) {
+            NSData *ed = decode(maybeEd25519);
+            assert(32 == sizeof(self->ed25519_public_key));
+            [ed getBytes:self->ed25519_public_key length:sizeof(self->ed25519_public_key)];
+        }
+    }
+    return self;
+}
+
+- (BOOL) isEqualToKey:(SUPublicKeys *)key {
+    NSString *thisKey = self.dsaPubKey;
+    NSString *thatKey = key.dsaPubKey;
+    if (thisKey == nil || thatKey == nil) {
+        return NO;
+    }
+    return NO;
+}
```

Comment:
Side note: I don't understand this method at all :p (I'm aware it's in 1.x)

Reply:
Hahahahaha, yeah, me either. This may as well be a plain `return NO;`, given the current logic. 

I _assume_ that there is supposed to be a comparison between the two DSA public keys there?

Reply:
Oh boy, that's awful. It's a public method in theory but removing it is probably the safest thing, isn't it.

Reply:
I marked it unavailable and implemented `-isEqual:` correctly. Think that's a good path forward?

Reply:
It looks like a refactoring issue from 5139179487d45fa3cda50ebeea26d67a19e10cba . I presume it's harmless as it's unused but @kornelski may have more context.

Reply:
It's OK to remove stuff that looks unused.

The original code for this was pretty convoluted. I've tried to simplify it, replacing boolean flags with control flow. If it's still bad, feel free to chop it further.

Reply:
Removed it completely. We can always add it back or call it `-isEqual:` if it turns out people were using this.